File: typescript/src/ne/segtable.ts
import ByteStream from "../bytestream";

export class SegmentTable {
    SegmentDataOffset: number;
    SegmentLength: number;
    Flags: number;
    MinimumSegmentAllocationSize: number;

    public constructor(segDataOff: number, segLen: number, flags: number, minSegAllocSize: number) {
        this.SegmentDataOffset = segDataOff;
        this.SegmentLength = segLen;
        this.Flags = flags;
        this.MinimumSegmentAllocationSize = minSegAllocSize;
    }

    public static Read(bst: ByteStream) {
        return new SegmentTable(bst.ReadWord(), bst.ReadWord(), bst.ReadWord(), bst.ReadWord());
    }
}

File: rust/src/executable/mz/ne/segtable.rs
use crate::byte_stream::ByteStream;

pub struct SegmentTable {
    pub data_offset: u16,
    pub length: u16,
    pub flags: u16,
    pub min_alloc_size: u16,
}

impl SegmentTable {
    pub fn read(bst: &mut ByteStream) -> Self {
        Self {
            data_offset: bst.read_word(),
            length: bst.read_word(),
            flags: bst.read_word(),
            min_alloc_size: bst.read_word(),
        }
    }
}
